composeml/label_times.py

```diff
@@ -226,3 +226,38 @@ def bin(self, bins, quantiles=False, labels=None, right=True):
 
         label_times.transforms.append(transform)
         return label_times
+
+    def sample(self, n=None, frac=None, random_state=None):
+        if isinstance(n, int):
+            sample = super().sample(n=n, random_state=random_state)
+            return sample
+
+        if isinstance(n, dict):
+            sample_per_label = []
+            for label, n, in n.items():
+                is_label = self[self.name].eq(label)
+                label = self.loc[self.index[is_label]]
+                sample = label.sample(n=n, random_state=random_state)
+                sample_per_label.append(sample)
+
+            labels = pd.concat(sample_per_label, axis=0, sort=False)
+            return labels
+
+        if isinstance(frac, float):
+            n = int(len(self) * frac)
+            sample = super().sample(n=n, random_state=random_state)
```

Comment:
pandas `sample` has a `frac` parameter: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sample.html

can we use that?

Reply:
yes, refactored to use pandas parameter